File: library/src/test/java/me/proxer/library/enums/MediaLanguageTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class MediaLanguageTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final MediaLanguage mediaLanguage = api.moshi().adapter(MediaLanguage.class).fromJson("\"de\"");

        assertThat(mediaLanguage).isSameAs(MediaLanguage.GERMAN);
    }

    @Test
    void testFallback() throws IOException {
        final MediaLanguage mediaLanguage = api.moshi().adapter(MediaLanguage.class).fromJson("\"xyz\"");

        assertThat(mediaLanguage).isSameAs(MediaLanguage.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/MediaLanguageTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class MediaLanguageTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(MediaLanguage::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"de\"") shouldBe MediaLanguage.GERMAN
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe MediaLanguage.OTHER
    }
}
